src/android/Capture.java

```diff
@@ -287,20 +291,59 @@ private static void createWritableFile(File file) throws IOException {
         file.setWritable(true, false);
     }
 
+    private String[] getRequiredPermissions(String[] allPerms) {
+        // not allowed to do multiple individual permissions requests so returning an array
+        ArrayList<String> neededPerms = new ArrayList<String>(allPerms.length);
+        for (String perm : allPerms) {
+            if (!PermissionHelper.hasPermission(this, perm)) {
+                neededPerms.add(perm);
+            }
+        }
+        return neededPerms.toArray(new String[]{});
+    }
+
     /**
      * Sets up an intent to capture video.  Result handled by onActivityResult()
      */
     private void captureVideo(Request req) {
-        if(cameraPermissionInManifest && !PermissionHelper.hasPermission(this, Manifest.permission.CAMERA)) {
-            PermissionHelper.requestPermission(this, req.requestCode, Manifest.permission.CAMERA);
+        String[] neededPermissions = this.getRequiredPermissions(new String[] {
+                Manifest.permission.READ_EXTERNAL_STORAGE,
+                Manifest.permission.CAMERA,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE});
+        if (neededPermissions.length > 0) {
+            PermissionHelper.requestPermissions(this, req.requestCode, neededPermissions);
         } else {
             Intent intent = new Intent(android.provider.MediaStore.ACTION_VIDEO_CAPTURE);
 
-            if(Build.VERSION.SDK_INT > 7){
-                intent.putExtra("android.intent.extra.durationLimit", req.duration);
-                intent.putExtra("android.intent.extra.videoQuality", req.quality);
+            final ContentResolver contentResolver = this.cordova.getActivity().getContentResolver();
+            final ContentValues cv = new ContentValues();
```

Comment:
not a fan of the short variable name

Reply:
Yeah I took that from elsewhere in this file where they do the same thing for images. Can change it